src/Board/Board.php

```diff
@@ -18,20 +21,32 @@ public function __construct()
         $rows    = static::getRows();
         $white   = true;    // Is the tile white? We'll flip this to false to mean black.
         
-        foreach ($columns as $column) {
-            foreach ($rows as $row) {
+        foreach ($rows as $row) {
+            foreach ($columns as $column) {
                 $colour    = $white ? 'white' : 'black';
-                $reference = $column . $row;
+                $reference = $row . $column;
                 $tile      = new Tile($colour, $reference);
                 
-                if (2 === $row) {
+                if (2 === $column) {
                     $tile->setPiece(new Pawn('white'));
-                } elseif (7 === $row) {
+                } elseif (7 === $column) {
                     $tile->setPiece(new Pawn('black'));
                 } elseif (('C1' === $reference) || ('F1' === $reference)) {
                     $tile->setPiece(new Bishop('white'));
                 } elseif (('C8' === $reference) || ('F8' === $reference)) {
                     $tile->setPiece(new Bishop('black'));
+                } elseif (('A1' === $reference) || ('H1' === $reference)){
+                    $tile->setPiece(new Rook('white'));
+                } elseif (('A8' === $reference) || ('H8' === $reference)){
+                    $tile->setPiece(new Rook('black'));
+                } elseif ('E1' === $reference) {
+                    $tile->setPiece(new King('white'));
+                } elseif ('E8' === $reference) {
+                    $tile->setPiece(new King('black'));
+                } elseif ('D1' === $reference){
```

Comment:
Added but I think $colour referred to the tile colour rather than piece colour previously so had to add an extra statement to account for that.